numpyro/contrib/tfp/distributions.py

```diff
@@ -235,21 +248,11 @@ def _onehot_enumerate_support(self, expand=True):
         continue
 
     try:
-        _PyroDist = locals()[_name]
-    except KeyError:
         _PyroDist = TFPDistribution[_Dist]
-        _PyroDist.__module__ = __name__
-        if hasattr(numpyro_dist, _name):
-            numpyro_dist_class = getattr(numpyro_dist, _name)
-            # resolve FooProbs/FooLogits namespaces
-            if type(numpyro_dist_class).__name__ == "function":
-                if not hasattr(numpyro_dist, _name + "Logits"):
-                    continue
-                numpyro_dist_class = getattr(numpyro_dist, _name + "Logits")
-            _PyroDist.arg_constraints = numpyro_dist_class.arg_constraints
-            _PyroDist.has_enumerate_support = numpyro_dist_class.has_enumerate_support
-            _PyroDist.enumerate_support = numpyro_dist_class.enumerate_support
-        locals()[_name] = _PyroDist
+    except ValueError:
```

Comment:
Can this ever actually be hit? I included it to replace the `continue` statement in the original code, but I think the preceding if statements mean it's unnecessary?

Reply:
Because you moved all the dispatching logic to the `__getitem__`, this is not required. Please feel free to remove this.